src/Google/GlobalSiteTag.php

```diff
@@ -113,14 +113,20 @@ public function register(): void {
 		$ads_conversion_label = $conversion_action['conversion_label'];
 
 		if ( $this->is_gtg_enabled() ) {
-			$gtg_adapter = Adapter::create();
-			$gtg_adapter->update(
-				[
-					'tagId'           => $conversion_action['conversion_id'],
-					'measurementPath' => '/wc/google/metrics/',
-				]
+			add_action(
+				'init',
+				function () use ( $conversion_action ) {
```

Comment:
As the plugin didn't follow it but in new improvement we should use `static` for clauses. WDYT?

```suggestion
				static function () use ( $conversion_action ) {
```

Reply:
Possibly so! This PR didn't mean to include these changes, so this could be reverted if @jjgrainger and @ankitrox want to use this PR, otherwise, feel free to close this PR if you are going to handle the upgrade directly in #3111.

Reply:
@joemcgill @mukeshpanchal27 I've updated the PR to revert the changes here as these were made to test and share potential solution and fall outside of the scope of the ticket (updating the library files).

I will mark this as resolve as we will tackle this as part of a different PR. Thanks